spring-cloud-contract-tools/spring-cloud-contract-gradle-plugin/src/main/groovy/org/springframework/cloud/contract/verifier/plugin/GenerateClientStubsFromDslTask.groovy

```diff
@@ -16,66 +16,51 @@
 import org.springframework.cloud.contract.verifier.converter.RecursiveFilesConverter
 
 //TODO: Implement as an incremental task: https://gradle.org/docs/current/userguide/custom_tasks.html#incremental_tasks ?
 /**
- * Generates stubs from the contracts. The name is WireMock related but the implementation
- * can differ
+ * Generates stubs from the contracts.
  *
  * @since 2.0.0
  */
 @CompileStatic
-class GenerateClientStubsFromDslTask extends ConventionTask {
+class GenerateClientStubsFromDslTask extends DefaultTask {
 
-	File stubsOutputDir
+	static final String TASK_NAME = 'generateClientStubs'
+	private static final String DEFAULT_MAPPINGS_FOLDER = 'mappings'
+
+	@InputDirectory
+	Provider<Directory> contractsDslDir
+	@Input
+	List<String> excludedFiles
+	@Input
+	boolean excludeBuildFolders
 
-	ContractVerifierExtension configProperties
-	GradleContractsDownloader downloader
+	@OutputDirectory
+	File stubsOutputDir
 
 	@TaskAction
 	void generate() {
-		logger.info("Stubs output dir [${getStubsOutputDir()}")
-		Task copyContractsTask = project.getTasksByName(SpringCloudContractVerifierGradlePlugin.COPY_CONTRACTS_TASK_NAME, false).first()
-		ContractVerifierConfigProperties props = props(copyContractsTask)
-		File contractsDslDir = contractsDslDir(copyContractsTask, props)
+		logger.info("Stubs output dir [${stubsOutputDir}")
 		logger.info("Spring Cloud Contract Verifier Plugin: Invoking DSL to client stubs conversion")
-		props.contractsDslDir = contractsDslDir
-		props.includedContracts = ".*"
-		File outMappingsDir = OutputFolderBuilder.outputMappingsDir(project, getStubsOutputDir())
-		logger.info("Contracts dir is [${contractsDslDir}] output stubs dir is [${outMappingsDir}]")
-		RecursiveFilesConverter converter = new RecursiveFilesConverter(props, outMappingsDir)
+		logger.info("Contracts dir is [${contractsDslDir}] output stubs dir is [${stubsOutputDir}]")
+		RecursiveFilesConverter converter = new RecursiveFilesConverter(stubsOutputDir, contractsDslDir.get().asFile, excludedFiles, ".*", excludeBuildFolders)
 		converter.processFiles()
 	}
 
-	@CompileDynamic
-	private ContractVerifierConfigProperties props(Task task) {
-		try {
-			return task.ext.contractVerifierConfigProperties
-		}
-		catch (Exception e) {
-			project.logger.error("Couldn't retrieve the configuration property set by the copy contracts task", e)
-			ContractVerifierConfigProperties props = ExtensionToProperties.fromExtension(getConfigProperties())
-			getDownloader().downloadAndUnpackContractsIfRequired(getConfigProperties(), props)
-			return props
-		}
-	}
-
-	@CompileDynamic
-	private File contractsDslDir(Task task, ContractVerifierConfigProperties props) {
-		try {
-			return task.ext.contractsDslDir
-		}
-		catch (Exception e) {
-			project.logger.error("Couldn't retrieve the contract dsl property set by the copy contracts task", e)
-			return props.contractsDslDir
-		}
+	static File createTaskOutput(String root, File stubsOutputDir, Project project) {
```

Comment:
This appears unused.

Reply:
The same here. Used in the `SpringCloudContractVerifierGradlePlugin`